app/services/search_service/exact_search.rb

```diff
@@ -58,7 +58,13 @@ def find_commodity(query)
 
       # NOTE: at the moment scope .declarable is not enough to
       # determine if it is really declarable or not
-      commodity.present? && commodity.declarable? ? commodity : nil
+      if commodity.blank?
+        nil
```

Comment:
this looks like a good case for clause guard:

`return nil unless commodity.present?`